src/main/groovy/com/bmuschko/gradle/docker/utils/MainClassFinder.groovy

```diff
@@ -138,13 +138,13 @@ class MainClassFinder {
                 pushAllSorted(stack, file.listFiles(new FileFilter() {
                     @Override
                     boolean accept(File pathname) {
-                        file.isDirectory() && !file.getName().startsWith(".")
+                        pathname.isDirectory() && !pathname.getName().startsWith(".")
                     }
                 }))
                 pushAllSorted(stack, file.listFiles(new FileFilter() {
                     @Override
                     boolean accept(File pathname) {
-                        file.isFile() && file.getName().endsWith(DOT_CLASS)
+                        pathname.isFile() && pathname.getName().endsWith(DOT_CLASS)
```

Comment:
This part should be the fix. The filter on class files is already supposed to be in the second pushAllSorted method.
But this doesn't work correctly because in current code in master the accept method checks file instead of pathname, therefore the first accept method accepts everything as long as file doesn't start with '.' (file is always a directory here. So if this part is fixed, al files on the stack are already filtered by a .class extension, so the other check is unnecessary.

Reply:
Send in another PR?

Reply:
Send in another PR?